pipeline/backend/kubernetes/image_pull_secret.go

```diff
@@ -0,0 +1,57 @@
+package kubernetes
+
+import (
+	"encoding/json"
+
+	"github.com/distribution/reference"
+	config_file "github.com/docker/cli/cli/config/configfile"
+	config_file_types "github.com/docker/cli/cli/config/types"
+	"go.woodpecker-ci.org/woodpecker/v2/pipeline/backend/types"
+	"go.woodpecker-ci.org/woodpecker/v2/pipeline/frontend/yaml/utils"
+	v1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func needsImagePullSecret(step *types.Step) bool {
+	return step.AuthConfig.Username != "" && step.AuthConfig.Password != ""
+}
+
+func mkImagePullSecret(step *types.Step, config *config, podName, goos string, options BackendOptions) (*v1.Secret, error) {
+	labels, err := podLabels(step, config, options)
+	if err != nil {
+		return nil, err
+	}
+	annotations := podAnnotations(config, options)
```

Comment:
Why do you use _Pod_ annotations on secret?

Reply:
Probably, applied to labels also.

Reply:
I assumed the user would want to label and annotate the secrets as well, but if you prefer a second pair of env vars for the secrets, I can do that.

Reply:
Added 4 new env's for labels, annotations, and allow for both for registry secrets. Added separate func's to return labels and annotations specifically for registry secrets.